Src/ILGPU.Algorithms.Tests/Generic/AlgorithmsTestData.tt

```diff
@@ -0,0 +1,575 @@
+    }
+
+    internal readonly struct UInt64ToBitCompInt64Transformer
+        : ITransformer<ulong, long>, IXunitSerializable
+    {
+        public long Transform(ulong value) => (long)~value;
+
+        public void Deserialize(IXunitSerializationInfo info) { }
+
+        public void Serialize(IXunitSerializationInfo info) { }
+    }
+
+<#  
+    foreach (var type in types) {
+#>
+    internal readonly struct <#= type.Name #>Mul2
+        : ITransformer<<#= type.Type #>, <#= type.Type #>>, IXunitSerializable
+    {
+        public <#= type.Type #> Transform(<#= type.Type #> value) => value * 2;
+
+        public void Deserialize(IXunitSerializationInfo info) { }
+
+        public void Serialize(IXunitSerializationInfo info) { }
+    }
+<#
+    }
+#>
+
+<#  
+    foreach (var type in types) {
+#>
+    internal readonly struct Index1To<#= type.Name #>Transformer 
+        : ITransformer<Index1, <#= type.Type #>>, IXunitSerializable
+    {
+<#
+        if(type.IsUnsignedInt) {
+#>
+        public <#= type.Type #> Transform(Index1 value) => (<#= type.Type #>)value;
+<#
+        }else{
+#>
+        public <#= type.Type #> Transform(Index1 value) => value;
+<#
+        }
+#>
+
+        public void Deserialize(IXunitSerializationInfo info) { }
+
+        public void Serialize(IXunitSerializationInfo info) { }
+    }
+<#
+    }
+#>
+
+    #endregion
+
+    #region Xunit RadixSortOperations Structures
+
+<#  
+    foreach (var type in IntTypes) { 
+#>
+    internal readonly struct XunitAscending<#=type.Name #>
+        : IRadixSortOperation<<#= type.Type #>>, IXunitSerializable
+    {
+        public int NumBits => sizeof(<#= type.Type #>) * 8;
+
+        public <#= type.Type #> DefaultValue => 0;
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public int ExtractRadixBits(<#= type.Type #> value, int shift, int bitMask) =>
+            (int)(value >> shift) & bitMask;
+
+        public void Deserialize(IXunitSerializationInfo info) { }
+
+        public void Serialize(IXunitSerializationInfo info) { }
+    }
+
+    internal readonly struct XunitDescending<#= type.Name #>
+        : IRadixSortOperation<<#= type.Type #>>, IXunitSerializable
+    {
+        public int NumBits => sizeof(<#= type.Type #>) * 8;
+
+        public <#= type.Type #> DefaultValue => 0;
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public int ExtractRadixBits(<#= type.Type #> value, int shift, int bitMask)
+        {
+            Ascending<#= type.Name #> operation = default;
+            return (~operation.ExtractRadixBits(value, shift, bitMask)) & bitMask;
+        }
+
+        public void Deserialize(IXunitSerializationInfo info) { }
+
+        public void Serialize(IXunitSerializationInfo info) { }
+    }
+
+<#  
+    }
+#>
+    #endregion
+
+    #region Xunit Vectors
+
+    #region Vector2
+
+    internal readonly struct Vector2Zero : IVector<Vector2>
+    {
+        Vector2 IVector<Vector2>.GetVector() => Vector2.Zero;
```

Comment:
Question: can we make these implementation `public` instead of an explicit interface implementation?